gateway/api_loader.go

```diff
@@ -561,14 +562,14 @@ func (d *DummyProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		var handler http.Handler
 		if r.URL.Hostname() == "self" {
 			if h, found := d.Gw.apisHandlesByID.Load(d.SH.Spec.APIID); found {
-				handler = h.(http.Handler)
+				handler = h.(*ChainObject).ThisHandler
```

Comment:
Can you please explain the reasoning for this change? It's looking like a fix for an impossible cast? Maybe add some error checking, which would surface the problem before? `if chain, ok := h.(*ChainObject); ok { handler = chain.ThisHandler } else { log... }`?